app/javascript/packs/event_receivers.jsx

```diff
@@ -34,41 +36,93 @@ class EventReceivers extends React.Component {
       .catch(e => console.error(e));
   }
 
-  saveReceiver(id, data) {
-    console.log(id, data);
-    const token = document.querySelector('meta[name="csrf-token"]').getAttribute('content');
-    console.log(token);
-    fetch(`/event_receivers/${id}.json`, {
-      method: 'PATCH',
+  getToken() {
+    return document.querySelector('meta[name="csrf-token"]').getAttribute('content');
+  }
+
+  saveReceiver(index, data) {
+    const id = this.state.event_receivers[index].id;
+    let url, method;
+    console.log(index);
+
+    if (id === null) {
```

Comment:
Can you create a separate helper function for this logic, and then return the tuple of `[url, method]` using [destructing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) please?

ie.

```js
function getURLandHTTPmethod(id) {
  if (id === null) {
    return ['/event_receivers', 'POST'];
  } else {
    return [`/event_receivers/${id}.json`, 'PATCH'];
  }
}

…

let [url, method] = getURLandHTTPmethod(id);
```

Reply:
I think it's a bit easier to reason about with regards to function scope.

Reply:
At first I was like. Hmmm, why is Dave not defining `url` and `method`? 🤔

Reply:
Some I only use it here, mind if I destructure assign from a ternary instead of a function? Probably will look ok with decent spacing

Reply:
*since

Reply:
Sure, lol, true that